dask_planner/src/expression.rs

```diff
@@ -529,152 +529,206 @@ impl PyExpr {
         match &self.expr {
             Expr::Literal(scalar_value) => match scalar_value {
-                ScalarValue::UInt32(iv) => Ok(iv.unwrap()),
+                ScalarValue::UInt32(iv) => Ok(*iv),
                 _ => Err(py_type_err("getValue<T>() - Unexpected value")),
             },
             _ => Err(py_type_err("getValue<T>() - Non literal value encountered")),
         }
     }
 
     #[pyo3(name = "getUInt64Value")]
-    pub fn uint_64_value(&mut self) -> PyResult<u64> {
+    pub fn uint_64_value(&mut self) -> PyResult<Option<u64>> {
         match &self.expr {
             Expr::Literal(scalar_value) => match scalar_value {
-                ScalarValue::UInt64(iv) => Ok(iv.unwrap()),
+                ScalarValue::UInt64(iv) => Ok(*iv),
+                _ => Err(py_type_err("getValue<T>() - Unexpected value")),
+            },
+            _ => Err(py_type_err("getValue<T>() - Non literal value encountered")),
+        }
+    }
+
+    #[pyo3(name = "getDate32Value")]
+    pub fn date_32_value(&mut self) -> PyResult<Option<i32>> {
+        match &self.expr {
+            Expr::Literal(scalar_value) => match scalar_value {
+                ScalarValue::Date32(iv) => Ok(*iv),
+                _ => Err(py_type_err("getValue<T>() - Unexpected value")),
+            },
+            _ => Err(py_type_err("getValue<T>() - Non literal value encountered")),
+        }
+    }
+
+    #[pyo3(name = "getDate64Value")]
+    pub fn date_64_value(&mut self) -> PyResult<Option<i64>> {
+        match &self.expr {
+            Expr::Literal(scalar_value) => match scalar_value {
+                ScalarValue::Date64(iv) => Ok(*iv),
+                _ => Err(py_type_err("getValue<T>() - Unexpected value")),
+            },
+            _ => Err(py_type_err("getValue<T>() - Non literal value encountered")),
+        }
+    }
+
+    #[pyo3(name = "getTime64Value")]
+    pub fn time_64_value(&mut self) -> PyResult<Option<i64>> {
+        match &self.expr {
+            Expr::Literal(scalar_value) => match scalar_value {
+                ScalarValue::Time64(iv) => Ok(*iv),
+                _ => Err(py_type_err("getValue<T>() - Unexpected value")),
+            },
+            _ => Err(py_type_err("getValue<T>() - Non literal value encountered")),
+        }
+    }
+
+    #[pyo3(name = "getTimestampValue")]
+    pub fn timestamp_value(&mut self) -> PyResult<(Option<i64>, Option<String>)> {
+        match &self.expr {
+            Expr::Literal(scalar_value) => match scalar_value {
+                ScalarValue::TimestampNanosecond(iv, tz)
+                | ScalarValue::TimestampMicrosecond(iv, tz)
+                | ScalarValue::TimestampMillisecond(iv, tz)
+                | ScalarValue::TimestampSecond(iv, tz) => {
+                    if let Some(iv) = iv {
+                        println!("{}", iv);
+                    }
+                    if let Some(tz) = tz {
+                        println!("{}", tz);
+                    }
```

Comment:
If we want to keep this debug logging, we should use `trace!` rather than `println!`

Reply:
nice catch. accidentally left it in